qiskit_nature/second_q/properties/occupied_modals.py

```diff
@@ -14,50 +14,29 @@
 
 from __future__ import annotations
 
-from typing import Optional, Mapping
+from typing import Mapping, Sequence
 
 import qiskit_nature  # pylint: disable=unused-import
 from qiskit_nature.second_q.operators import VibrationalOp
 
-from .bases import VibrationalBasis
-
 
 class OccupiedModals:
     """The OccupiedModals property."""
 
-    def __init__(
-        self,
-        basis: Optional[VibrationalBasis] = None,
-    ) -> None:
+    def __init__(self, num_modals: Sequence[int]) -> None:
         """
         Args:
-            basis: the
-                :class:`~qiskit_nature.second_q.properties.bases.VibrationalBasis`
-                through which to map the integrals into second quantization. This attribute **MUST**
-                be set before the second-quantized operator can be constructed.
+            num_modals: the number of modals for each mode.
         """
-        self._basis: VibrationalBasis = basis
-
-    @property
-    def basis(self) -> VibrationalBasis:
-        """Returns the basis."""
-        return self._basis
-
-    @basis.setter
-    def basis(self, basis: VibrationalBasis) -> None:
-        """Sets the basis."""
-        self._basis = basis
+        self.num_modals = num_modals
```

Comment:
Attribute: docs for this I think - unless you want a read-only property versus something that can be updated (written) too.